### ISSUE TITLE
Category - Notice on incorrect price filter GET param

### ISSUE THREAD

**Issue by ihor-sviziev** [magento/magento2#25911]
### Preconditions (*)
1. Magento 2.3.2-p2
2. Single category with few products

### Steps to reproduce (*)
1. Go to Category page that has filters by price
2. To the end of URL add `?price=-100%27[0]`, for instance https://example.com/my-category/?price=-100%27[0]
3. See logs

### Expected result (*)
1. We should not have error logs
2. Layered navigation block should be shown

### Actual result (*)
1. We have following logs in system.log:
```
[2019-12-05 11:25:41] main.CRITICAL: Notice: A non well formed numeric value encountered in /var/www/magento2/vendor/magento/module-catalog-search/Model/Layer/Filter/Price.php on line 180 [] []
[2019-12-05 11:25:43] main.CRITICAL: The "catalog.leftnav" block threw an exception, and it can't be rendered. [] []
```
2. Layered navigation block isn't shown

**Note:** this value isn't passed to DB, so _no security issue there_

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @krishprakash 
 Thank you for verifying the issue. Based on the provided information internal tickets `MC-29531` were created

**Issue Available**: @krishprakash, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by magento-engcom-team** [comment]
Hi @ihor-sviziev. Thank you for your report. 
The issue has been fixed in magento/magento2#25912 by @ihor-sviziev in 2.4-develop branch
Related commit(s):
 - [706826899aa6c9452938d8d241b51106caa51633](https://github.com/magento/magento2/commit/706826899aa6c9452938d8d241b51106caa51633)
 - [414105e20221688f94e45b0563f1b38f9e9ea258](https://github.com/magento/magento2/commit/414105e20221688f94e45b0563f1b38f9e9ea258)

The fix will be available with the upcoming 2.4.0 release.
